src/App.js

```diff
@@ -1,26 +1,143 @@
-import React from 'react';
-import logo from './logo.svg';
+/*
+ * Main App.
+ */
+
+import React, { Component } from 'react';
+
+// Bulma and custom CSS.
+import 'bulma/css/bulma.css';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+// Data.
+import foods from './foods.json';
+import FoodBoxList from './components/FoodBoxList';
+
+// Components.
+import AddFood from './components/AddFood';
+
+class App extends Component {
+  state = {
+    foodList: foods,
+    searchResult: foods,
+    showForm: false,
+    foodDetails: [],
+    totalCalories: [],
+  };
+
+  handleAdd = (event) => {
+    event.preventDefault();
+    const { foodList, searchResult } = this.state;
+    const { name, calories, image } = event.target;
+
+    const newFood = {
+      name: name.value,
+      calories: calories.value,
+      image: image.value,
+    };
+
+    this.setState({
+      foodList: [...foodList, newFood],
+      searchResult: [...searchResult, newFood],
+      showForm: false,
+    });
+  };
+
+  handleForm = () => {
+    this.setState({
+      showForm: true,
+    });
+  };
+
+  handleSearch = (event) => {
+    const searchText = event.target.value;
```

Comment:
you may also want to apply toLowerCase to the search text, to improve the search functionality